static/js/hotkey.js

```diff
@@ -1001,6 +1002,29 @@ export function process_hotkey(e, hotkey) {
             stream_popover.build_move_topic_to_stream_popover(msg.stream_id, msg.topic, msg);
             return true;
         }
+        case "zoom_to_message_near": {
+            switch (msg.type) {
+                case "private":
+                    narrow.activate(
+                        [
+                            {operator: "pm-with", operand: msg.reply_to},
+                            {operator: "near", operand: msg.id},
+                        ],
+                        {trigger: "hotkey"},
+                    );
+                    return true;
+                case "stream":
+                    narrow.activate(
+                        [
+                            {operator: "stream", operand: msg.stream},
+                            {operator: "topic", operand: msg.topic},
+                            {operator: "near", operand: msg.id},
+                        ],
+                        {trigger: "hotkey"},
+                    );
+                    return true;
+            }
+        }
```

Comment:
I think this is equivalent to doing `window.location.hash = by_conversation_and_time_url(message)`. Maybe it's worth a comment noting that -- I think we prefer to do this because we want to pass the `trigger` parameter, for example.

Reply:
Yes, the effect is the same (manually tested). Added a comment!

Reply:
@timabbott Simply using `window.location = hashutil.by_conversation_and_time_url(msg)` solves the issue of #24468, though that means we don't pass in the `trigger` parameter. That sounds like an acceptable trade-off to me though. Thoughts?

Reply:
I think that's probably OK, we should just leave a comment on why we did this. We aren't currently using the `trigger` logging, so we mostly will just want to make sure we know to figure out how to add it back if we want it.

Reply:
That said, we could also look at whether having `set_hash` check if we already have that hash would fix this better.

Reply:
Thanks for the suggestions! I've implemented the `set_hash` fix in #24611 -- I think that's a better, more generalised and future proof solution